### DISCUSSION TITLE
Tax calculations

### DISCUSSION THREAD

**Discussion by nUpl** [coreshop/CoreShop#2333] [Q&A]
Hey,
I've been having some issues regarding tax calculations in carts.
The prices in our shop are set up to already include taxes.

Example 1:
When I put an Item in my cart that costs 186€, use a quantity of 2, with a tax rate of 19%, and no shipping, Coreshop calulates a tax of 59.40€.
According to what I gathered from `getTaxesAmountFromGrossAsArray` in `CoreShop\src\CoreShop\Component\Taxation\Calculator\TaxRulesTaxCalculator.php` the way CoreShop calculates the tax in this case is
tax = gross - net
tax = 372 - (372 / (1 + (19 / 100))) = 59.3949€, which when rounded would be 59.39€ instead of 59.40€, if I'm not mistaken.

Example 2:
However, if I have an item, which costs 183.50€, with quantitiy of 2, same tax rate 19%, but shipping of 5€ Coreshop calculates a tax of 59.39€, even though this equals to the same gross price = 183.50€ * 2 + 5€ = 372€

Can you reproduce this? Any help is appreciated.

Disclosure: I don't have much experience in accounting, so I might be wrong somewhere.

I've also seen #1849 , which may be related.

**Comment by dpfaffenbauer** [comment]
Tax always depends on how you calculate it. There are several different valid ways of doing it. Can you give me a concrete example of what you expect to get and what CoreShop calculates?

**Reply by nUpl** [reply]
After some more examination the issue seems to be related to the precision setting we are using and not an error with coreshops calculations.

When we have an item costing 186€ with 19% tax coreshop currently shows a tax amount of 59.40€ and a total net price of 312.61€. With 59.4 + 312.61 = 372.01 this is actually 1 cent over the actual total. We will try changing our precision setting and see if that helps.
